pkg/sql/alter_index_visible.go

```diff
@@ -40,9 +85,43 @@ func (p *planner) AlterIndexVisible(
 // func (n *alterIndexVisibleNode) ReadingOwnWrites() {}
 
 func (n *alterIndexVisibleNode) startExec(params runParams) error {
-	return unimplemented.Newf(
-		"Not Visible Index",
-		"altering an index to visible or not visible is not supported yet")
+	if n.n.NotVisible && n.index.Primary() {
+		return pgerror.Newf(pgcode.FeatureNotSupported, "primary index cannot be invisible")
+	}
+
+	if n.index.IsNotVisible() == n.n.NotVisible {
+		// Nothing needed if the index is already what they want.
+		str := ""
+		if n.n.NotVisible {
+			str = " not"
+		}
+		// TODO (wenyihu6): should we log notices if index is already what they want or is that too noisy?
```

Comment:
cc @cockroachdb/sql-experience but I don't think so, if you do
```
CREATE TABLE t (i INT NOT NULL);
ALTER TABLE t ALTER COLUMN i SET NOT NULL;
```
that ALTER is entirely redundant yet there's no notice.